test/test_constraints.py

```diff
@@ -0,0 +1,63 @@
+import unittest
+import pytest
+import numpy as np
+from sklearn.utils import shuffle
+from metric_learn.constraints import Constraints
+
+SEED = 42
+
+
+def gen_labels_for_chunks(num_chunks, chunk_size,
+                          n_classes=10, n_unknown_labels=5):
+  """Generates num_chunks*chunk_size labels that split in num_chunks chunks,
+  that are homogeneous in the label."""
+  assert min(num_chunks, chunk_size) > 0
+  classes = shuffle(np.arange(n_classes), random_state=SEED)
+  n_per_class = chunk_size * (num_chunks // n_classes)
+  n_maj_class = chunk_size * num_chunks - n_per_class * (n_classes - 1)
+
+  first_labels = classes[0] * np.ones(n_maj_class, dtype=int)
+  remaining_labels = np.concatenate([k * np.ones(n_per_class, dtype=int)
+                                     for k in classes[1:]])
+  unknown_labels = -1 * np.ones(n_unknown_labels, dtype=int)
+
+  labels = np.concatenate([first_labels, remaining_labels, unknown_labels])
+  return shuffle(labels, random_state=SEED)
+
+
+class TestConstraints(unittest.TestCase):
+  def test_exact_num_points_for_chunks(self):
+    """Checks that the chunk generation works well with just enough points."""
+    for num_chunks, chunk_size in [(5, 10), (10, 50)]:
```

Comment:
Instead of having a loop here, you can maybe use `pytest.parametrize` (see https://github.com/scikit-learn-contrib/metric-learn/search?q=parametrize&unscoped_q=parametrize), I find it to make tests debugging easier (it creates sort of "subtests" that can be debugged independently e.g. with IDEs like PyCharm)
(though I'm not sure it will work here since we are in  a unittest class and sometimes they can interfere weirdly with pytest but you can always try if you want)

Reply:
Indeed `pytest.parametrize` is not compatible with the unittest class:
https://stackoverflow.com/questions/35561843/py-test-parametrize-test-cases-from-classes

I think using `pytest.parametrize` is indeed more elegant and modular, so I would suggest to remove the class and revert to the `pytest.parametrize` formulation you had in a previous commit. Or to use a regular class as in the first response in the link above so that both are compatible

Reply:
While using a class `TestConstraints` mimics the `Constraints` class in `constraints.py`, it is not necessary since the file is called `test_constraints.py`. I reverted to the `pytest.parametrize` formulation for brevity.

Reply:
> I think using `pytest.parametrize` is indeed more elegant and modular, so I would suggest to remove the class and revert to the `pytest.parametrize` formulation you had in a previous commit. Or to use a regular class as in the first response in the link above so that both are compatible

Ah yes that's right sorry @RobinVogel I hadn't seen you already had used `pytest.parametrize` in your previous commits
Yes I guess having tests with explicit names and explicit filenames should be enough for the structure, and pytest's modularity should make it easy to reorganize in the future if needed